ibm/service/kms/resource_ibm_kms_key_rings.go

```diff
@@ -153,7 +153,9 @@ func resourceIBMKmsKeyRingDelete(d *schema.ResourceData, meta interface{}) error
 	err = kpAPI.DeleteKeyRing(context.Background(), id[0], kp.WithForce(force_delete))
 	if err != nil {
 		kpError := err.(*kp.Error)
-		if kpError.StatusCode == 404 {
+		// Key ring deletion used to occur by silencing the 409 failed deletion and allowing instance deletion to clean it up
```

Comment:
suggestion:

For now, in order not to break existing Terraform customers, allowing Terraform DESTROY to succeeded even when the return from Key Ring deletion is HTTP 409.  That was behavior until IBM Terraform Cloud Provider 1.57.0. For more context see [Add description for keys and force_delete for deleteKeyRings for IBM KMS #4767](https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4767) and [Key Ring API spec](https://cloud.ibm.com/apidocs/key-protect#deletekeyring)